source/extensions/clusters/dynamic_forward_proxy/cluster.cc

```diff
@@ -158,15 +159,34 @@ Cluster::LoadBalancer::chooseHost(Upstream::LoadBalancerContext* context) {
             .getDataReadOnly<Router::StringAccessor>("envoy.upstream.dynamic_host");
   }
 
-  absl::string_view host;
+  absl::string_view raw_host;
   if (dynamic_host_filter_state) {
-    host = dynamic_host_filter_state->asString();
+    raw_host = dynamic_host_filter_state->asString();
   } else if (context->downstreamHeaders()) {
-    host = context->downstreamHeaders()->getHostValue();
+    raw_host = context->downstreamHeaders()->getHostValue();
   } else if (context->downstreamConnection()) {
-    host = context->downstreamConnection()->requestedServerName();
+    raw_host = context->downstreamConnection()->requestedServerName();
   }
 
+  const bool is_secure = cluster_.info()
+                             ->transportSocketMatcher()
+                             .resolve(nullptr)
+                             .factory_.implementsSecureTransport();
+  uint32_t port = is_secure ? 443 : 80;
+  if (context->downstreamConnection()) {
```

Comment:
To make sure I understand... We have a mechanism where an optional bit of state can be stored in envoy.upstream.dynamic_port, which tells us the actual upsteam port to us. That's what's going on here?

Maybe a tiny comment about envoy.upstream.dynamic_port? Or maybe it's already self-explanatory? Mostly I just want to make sure I understand this :)